clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -382,3 +382,13 @@ void CIRGenFunction::emitVAStartEnd(mlir::Value argValue, bool isStart) {
   else
     cir::VAEndOp::create(builder, argValue.getLoc(), argValue);
 }
+
+// FIXME(cir): This completely abstracts away the ABI with a generic CIR Op. We
+// need to decide how to handle va_arg target-specific codegen.
+mlir::Value CIRGenFunction::emitVAArg(VAArgExpr *ve, Address &vaListAddr) {
```

Comment:
The `vaListAddr` parameter isn't used, and it doesn't look like it will be. That does raise a question though. Classic codegen uses this parameter when emitting aggregate or complex va args to detect when the ABI didn't support the type for the arg. We won't detect that until a later lowering phase. Is that a problem? I don't think we have access to the same diagnostic reporting mechanism at that point.